src/app/clusters/scenes/SceneTableImpl.h

```diff
@@ -0,0 +1,244 @@
+            mAVPairs[pairCount].attributeID = aVPair.attributeID;
+            auto value_iterator             = aVPair.attributeValue.begin();
+
+            valueBytes = 0;
+            while (value_iterator.Next() && valueBytes < kMaxValueSize)
+            {
+                mValueBuffer[pairCount][valueBytes] = value_iterator.GetValue();
+                valueBytes++;
+            }
+            // Check we could go through all bytes of the value
+            VerifyOrReturnError(value_iterator.Next() == false, CHIP_ERROR_BUFFER_TOO_SMALL);
+            mAVPairs[pairCount].attributeValue = mValueBuffer[pairCount];
+            mAVPairs[pairCount].attributeValue.reduce_size(valueBytes);
+            pairCount++;
+        }
+
+        // Check we could go through all pairs in incomming command
+        VerifyOrReturnError(pair_iterator.Next() == false, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+        attributeValueList = mAVPairs;
+        attributeValueList.reduce_size(pairCount);
+
+        writer.Init(serialisedBytes);
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outer));
+        ReturnErrorOnFailure(app::DataModel::Encode(
+            writer, TLV::ContextTag(to_underlying(app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList)),
+            attributeValueList));
+        ReturnErrorOnFailure(writer.EndContainer(outer));
+
+        return CHIP_NO_ERROR;
+    }
+
+    /// @brief Simulates taking data from nvm and loading it in a command object if the cluster is supported by the endpoint
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serialisedBytes data to deserialize into EFS
+    /// @return CHIP_NO_ERROR if Extension Field Set was successfully populated, specific CHIP_ERROR otherwise
+    virtual CHIP_ERROR Deserialize(EndpointId endpoint, ClusterId cluster, const ByteSpan & serialisedBytes,
+                                   app::Clusters::Scenes::Structs::ExtensionFieldSet::Type & extensionFieldSet) override
+    {
+        app::DataModel::DecodableList<app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
+        app::Clusters::Scenes::Structs::AttributeValuePair::DecodableType decodePair;
+
+        TLV::TLVReader reader;
+        TLV::TLVType outer;
+        uint8_t pairCount  = 0;
+        uint8_t valueBytes = 0;
+
+        VerifyOrReturnError(SupportsCluster(endpoint, cluster), CHIP_ERROR_INVALID_ARGUMENT);
+
+        extensionFieldSet.clusterID = cluster;
+        reader.Init(serialisedBytes);
+        ReturnErrorOnFailure(reader.Next());
+        ReturnErrorOnFailure(reader.EnterContainer(outer));
+        ReturnErrorOnFailure(reader.Next());
+        attributeValueList.Decode(reader);
+
+        auto pair_iterator = attributeValueList.begin();
+        while (pair_iterator.Next() && pairCount < kMaxAvPair)
```

Comment:
Similar comments to the above: Up-front length checks would be better, need to check iterator status after loop.  Please see the documentation in DecodableList.h.